src/listen/websocket.rs

```diff
@@ -317,13 +384,22 @@ where
         };
         let (ws_stream, _) = tokio_tungstenite::connect_async(request).await?;
         let (write, mut read) = ws_stream.split();
-        let write = Arc::new(Mutex::new(write));
+        let write_arc = Arc::new(Mutex::new(write));
+        let cloned_write_arc = Arc::clone(&write_arc);
+        self.write_arc = Some(cloned_write_arc); // Store the write arc
         let (mut tx, rx) = mpsc::channel::<Result<StreamResponse>>(1);
 
+        let event_tx_open = event_tx.clone();
+        let event_tx_keep_alive = event_tx.clone();
+        let event_tx_send = event_tx.clone();
+        let event_tx_receive = event_tx.clone();
```

Comment:
Nit: This is fine, but I would create these clones when you need them, and just call each one `event_tx`, using the same shadowing pattern as in the `write_arc` comments above.

Reply:
Not sure I fully understood this but moved them to where they get used and only create the keep alive one if needed